checks/evaluation/signed_releases.go

```diff
@@ -17,117 +17,90 @@ package evaluation
 import (
 	"fmt"
 	"math"
-	"strings"
 
 	"github.com/ossf/scorecard/v4/checker"
 	sce "github.com/ossf/scorecard/v4/errors"
 	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/probes/releasesAreSigned"
+	"github.com/ossf/scorecard/v4/probes/releasesHaveProvenance"
 )
 
-var (
-	signatureExtensions  = []string{".asc", ".minisig", ".sig", ".sign"}
-	provenanceExtensions = []string{".intoto.jsonl"}
-)
-
-const releaseLookBack = 5
-
 // SignedReleases applies the score policy for the Signed-Releases check.
 // nolint
-func SignedReleases(name string, dl checker.DetailLogger, r *checker.SignedReleasesData) checker.CheckResult {
-	if r == nil {
-		e := sce.WithMessage(sce.ErrScorecardInternal, "empty raw data")
+func SignedReleases(name string,
+	findings []finding.Finding, dl checker.DetailLogger,
+) checker.CheckResult {
+	expectedProbes := []string{
+		releasesAreSigned.Probe,
+		releasesHaveProvenance.Probe,
+	}
+
+	if !finding.UniqueProbesEqual(findings, expectedProbes) {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "invalid probe results")
 		return checker.CreateRuntimeErrorResult(name, e)
 	}
 
-	totalReleases := 0
-	total := 0
-	score := 0
-	for _, release := range r.Releases {
-		if len(release.Assets) == 0 {
-			continue
+	// All probes have OutcomeNotApplicable in case the project has no 
+	// releases. Therefore, check for any finding with OutcomeNotApplicable.
+	for i := range findings {
+        f := &findings[i]
+	    if f.Outcome == finding.OutcomeNotApplicable {
+			dl.Warn(&checker.LogMessage{
+				Text: "no GitHub releases found",
+			})
+			// Generic summary.
+			return checker.CreateInconclusiveResult(name, "no releases found")
 		}
+	}
 
-		dl.Debug(&checker.LogMessage{
-			Text: fmt.Sprintf("GitHub release found: %s", release.TagName),
-		})
-
-		totalReleases++
-		signed := false
-		hasProvenance := false
-
-		// Check for provenance.
-		for _, asset := range release.Assets {
-			for _, suffix := range provenanceExtensions {
-				if strings.HasSuffix(asset.Name, suffix) {
-					dl.Info(&checker.LogMessage{
-						Path: asset.URL,
-						Type: finding.FileTypeURL,
-						Text: fmt.Sprintf("provenance for release artifact: %s", asset.Name),
-					})
-					hasProvenance = true
-					total++
-					break
+	score := 0
+	totalPositive := 0
+	totalReleases := 0
+	checker.LogFindings(findings, dl)
+
+	for i := range findings {
+        f := &findings[i]
+	    if f.Outcome == finding.OutcomePositive {
+	        switch f.Probe {
+	        case releasesAreSigned.Probe:
+	        	totalPositive++
```

Comment:
This logic is a bit confusing. Could we use something like a `map[int]int` or something to keep track of releases? We're iterating over `findings` here and in `releaseAlsoHasProvenance`.

Reply:
I don't fully understand: Which values should be in the map?

Is the purpose of this to know the score of each release? If yes, what is the purpose of knowing that? To log it?

Reply:
Yes, it seems like we're iterating twice. 

The map would keep track of scores per release. So you would have something like:
```
score[releaseID]releaseScore
for each finding:
    if the release has signature and map doesn't contain releaseID:
        score[finding.releaseID] = 8
    if the release has provenance:
        score[finding.releaseID] = 10
```

Is that the logic that you're working with?